assets/src/components/vehicleIcon.tsx

```diff
@@ -1,16 +1,299 @@
 import React from "react"
 
-const VehicleIcon = ({ scale }: { scale?: number }) => {
-  const scaleAmount = scale || 1.0
+export enum Orientation {
+  Up,
+  Down,
+  Left,
+  Right,
+}
+
+export enum Size {
+  Small,
+  Medium,
+  Large,
+}
+
+export interface Props {
+  size: Size
+  orientation: Orientation
+  label?: string
+  variant?: string | null
+}
+
+/*
+Size of the triangle for calculation of the viewbox and positioning the label
+Sizes are before scaling, relative to the center of the triangle
+*/
+const X_CENTER_TO_SIDE = 22
+const Y_CENTER_TO_POINT = 16
+const Y_CENTER_TO_BASE = 20
+
+export const VehicleIcon = ({ size, orientation, label, variant }: Props) => {
+  // shrink the viewbox to fit around the triangle and label
+  const scale = scaleForSize(size)
+  const labelBgWidth = label ? labelBackgroundWidth(size) : 0
+  const labelBgHeight = label ? labelBackgroundHeight(size) : 0
+  let left = 0
+  let right = 0
+  let top = 0
+  let bottom = 0
+  switch (orientation) {
```

Comment:
If you broke this logic out into component functions could you get rid of the mutable variables?

Reply:
You could, and I do that for some functions. For the ones that are left, I felt like the logic is pretty inseparable from the rest of the function.

I guess for this I could pull out `const viewBox(props: Props): {left: number, top: number, width: number, height: number}`. It won't get rid of the `let` but it'll help a bit.

Reply:
Yeah, if the let is inside a helper function I generally find that less worrisome.